src/C-DEngine/C-DCommunication/cdeQueuedSender/cde_QueuedSenderHB.cs

```diff
@@ -7,21 +7,35 @@
 using nsCDEngine.ISM;
 using nsCDEngine.ViewModels;
 using System;
+using System.Collections.Concurrent;
+using System.Linq;
 using System.Threading;
 
 namespace nsCDEngine.Communication
 {
     internal partial class TheQueuedSender
     {
         private readonly object lockStartLock = new ();
+        private Timer _tsmHistoryExpirationTimer = null;
+
         internal void StartHeartBeat()
         {
             if (IsQSenderReadyForHB || !TheBaseAssets.MasterSwitch)
                 return;
             IsQSenderReadyForHB = true;
+
             try
             {
-                MyTSMHistory ??= new TheMirrorCache<TheSentRegistryItem>(TheBaseAssets.MyServiceHostInfo.TO.QSenderDejaSentTime);
+                lock (lockStartLock)
+                {
+                    MyTSMHistory ??= new ConcurrentDictionary<string, TheSentRegistryItem>();
```

Comment:
Could you use cdeConcurrentDictionary instead of ConcurrentDictionary? There are platforms (i.e. eCLR) that do not support the ConcurrentDictionary and our cde... version does detect this and falls back to the regular Dictionary on those platforms. If ConcurrentDictionary is supported, we are using it natively in the cde... Version. Thanks

Reply:
Good hint - using the cdeConcurrentDictionary know.